llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -1398,6 +1780,79 @@ DebugAbbrevOffsetsRefAdaptor::encodeOffsets(ArrayRef<size_t> Offsets) {
   return EncodedOffsets;
 }
 
+static void
+getFixupsForCuDie(InMemoryCASDWARFObject::SplitUpCUData &SplitData,
+                  std::unordered_set<llvm::dwarf::Form> &FormsToFixup,
+                  DWARFDie &CUDie, MutableArrayRef<char> DebugInfoData,
+                  uint64_t &CUOffset, bool IsLittleEndian,
+                  uint8_t AddressSize) {
+
+  // Copy Abbrev Tag
+  DWARFDataExtractor DWARFExtractor(toStringRef(DebugInfoData), IsLittleEndian,
+                                    AddressSize);
+  auto PrevOffset = CUOffset;
+  DWARFExtractor.getULEB128(&CUOffset);
+  auto Size = CUOffset - PrevOffset;
+  for (unsigned I = 0; I < Size; I++)
+    SplitData.DebugInfoRefData.push_back(DebugInfoData[PrevOffset + I]);
+  for (const DWARFAttribute &AttrValue : CUDie.attributes()) {
+    if (FormsToFixup.find(AttrValue.Value.getForm()) != FormsToFixup.end()) {
+      for (unsigned I = 0; I < AttrValue.ByteSize; I++)
+        SplitData.FixupData.push_back(DebugInfoData[AttrValue.Offset + I]);
+    } else {
+      for (unsigned I = 0; I < AttrValue.ByteSize; I++)
+        SplitData.DebugInfoRefData.push_back(
+            DebugInfoData[AttrValue.Offset + I]);
+    }
+    CUOffset += AttrValue.ByteSize;
+  }
+
+  DWARFDie Child = CUDie.getFirstChild();
+  while (Child && Child.getAbbreviationDeclarationPtr()) {
+    getFixupsForCuDie(SplitData, FormsToFixup, Child, DebugInfoData, CUOffset,
+                      IsLittleEndian, AddressSize);
+    Child = Child.getSibling();
+  }
+  if (Child && !Child.getAbbreviationDeclarationPtr()) {
+    SplitData.DebugInfoRefData.push_back(DebugInfoData[CUOffset]);
+    CUOffset++;
+  }
+}
+
+InMemoryCASDWARFObject::SplitUpCUData
+InMemoryCASDWARFObject::identifyFixupsInCU(
```

Comment:
the naming is confusing here: identify doesn't sound like it actually performs the action, but it returns the split up data.